From ca082c475897661367ffe5c4bfc1dc4c7edfe707 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Fri, 21 Jun 2019 09:12:52 +0200
Subject: [PATCH 123/171] rtnetlink: annotate rtnl_unlock

SILENCES:

net/core/rtnetlink.c:117:1: warning: mutex 'rtnl_mutex' is still held at the end of function [-Wthread-safety-analysis]
}
^
net/core/rtnetlink.c:113:24: note: mutex acquired here
void rtnl_unlock(void) __releases_mutex(rtnl_mutex)
                       ^
./include/linux/compiler_attributes.h:267:56: note: expanded from macro '__releases_mutex'
\# define __releases_mutex(x)            __attribute__((release_capability(x)))
                                                       ^

Link: https://github.com/ClangBuiltLinux/thread-safety-analysis/issues/110
---
 net/core/rtnetlink.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/core/rtnetlink.c b/net/core/rtnetlink.c
index a9ee59374da7..3a980069a816 100644
--- a/net/core/rtnetlink.c
+++ b/net/core/rtnetlink.c
@@ -110,7 +110,7 @@ void __rtnl_unlock(void) __releases_mutex(rtnl_mutex)
 	}
 }
 
-void rtnl_unlock(void) __releases_mutex(rtnl_mutex)
+void rtnl_unlock(void) __releases_mutex(rtnl_mutex) __no_thread_safety_analysis
 {
 	/* This fellow will unlock it for us. */
 	netdev_run_todo();
-- 
2.17.1

